datafusion/functions-aggregate/src/approx_percentile_cont.rs

```diff
@@ -154,19 +154,20 @@ fn get_scalar_value(expr: &Arc<dyn PhysicalExpr>) -> Result<ScalarValue> {
 }
 
 fn validate_input_percentile_expr(expr: &Arc<dyn PhysicalExpr>) -> Result<f64> {
-    let percentile = match get_scalar_value(expr)? {
-        ScalarValue::Float32(Some(value)) => {
+    let percentile = match get_scalar_value(expr).ok() {
```

Comment:
Another way to express this for your consideration that might make the error handling more obvious is

```rust
      let scalar = match get_scalar_value(expr)
        .map_err(not_impl_err!("Percentile value for 'APPROX_PERCENTILE_CONT' must be a litera, got: {expr}l")?
      match scalar { 
          ScalarValue::Float32(Some(value)) => {
 ...
 ```

Reply:
It looks nice. Thanks for the suggestions!